src/agent/src/sandbox.rs

```diff
@@ -151,7 +151,12 @@ impl Sandbox {
     pub fn remove_sandbox_storage(&self, path: &str) -> Result<()> {
         let mounts = vec![path.to_string()];
         remove_mounts(&mounts)?;
-        fs::remove_dir_all(path).context(format!("failed to remove dir {:?}", path))?;
+        // "remove_dir" will fail if the mount point is backed by a read-only filesystem.
+        // This is the case with the device mapper snapshotter, where we mount the block device directly
+        // at the underlying sandbox path which was provided from the base RO kataShared path from the host.
```

Comment:
IMO we should mount the storage based file systems to a tmpfs location provided by the guest, thus it does not depend on `kataShared`. But let's fix it later on. Thanks for looking into the root cause!